qiskit/providers/options.py

```diff
@@ -12,10 +12,10 @@
 
 """Container class for backend options."""
 
-import types
+import io
 
 
-class Options(types.SimpleNamespace):
+class Options:
```

Comment:
It would also be good to add a `copy` method to this class.
Maybe also an iterator so you can do something like `for attr, value in options`?

Reply:
Sure I can add these, but is it ok in a follow up PR. I only modified options here like this to support the validators (so we had an api driven way to express bounds for options like what is contained in `backend.configuration()` for `BackendV1`). I'd like to avoid the feature creep too much in this PR (as it's already pushing 3000 lines).

Reply:
Sure, there is no reason those changes need to depend on this one, so you could even do it first if you wanted